python/ray/serve/_private/test_utils.py

```diff
@@ -699,3 +704,76 @@ def tlog(s: str, level: str = "INFO"):
 
     now = datetime.datetime.now().strftime("%H:%M:%S.%f")[:-3]
     print(f"[{level}] {now} {s}")
+
+
+def get_application_urls(
+    protocol: Union[str, RequestProtocol] = RequestProtocol.HTTP,
+    app_name: str = SERVE_DEFAULT_APP_NAME,
+    use_proxy_targets: bool = RAY_SERVE_USE_PROXY_TARGETS_FOR_GET_APPLICATION_URLS,
+) -> List[str]:
+    """Get the URL of the application.
+
+    Args:
+        protocol: The protocol to use for the application.
+        app_name: The name of the application.
+        use_proxy_targets: Whether to use the proxy targets.
+
+    Returns:
+        The URLs of the application.
+    """
+    client = _get_global_client()
+    serve_details = client.get_serve_details()
+    route_prefix = serve_details["applications"][app_name]["route_prefix"]
+    if use_proxy_targets:
+        target_groups: List[TargetGroup] = ray.get(
+            client._controller.get_proxy_target_groups.remote()
```

Comment:
as discussed offline, instead of introducing a new `get_proxy_target_groups`, we can have `get_target_groups` take an optional `app_name` parameter. cc @abrarsheikh